compiler.md

```diff
@@ -1,155 +1,483 @@
+  - [Baseline Performance Improved](#baseline-performance-improved)
+  - [New Language Features](#new-language-features)
+    - [New Language Features Support And Transpilers](#new-language-features-support-and-transpilers)
   - [Resources](#resources)
+- [Ignition Interpreter](#ignition-interpreter)
+- [Collecting Feedback via ICs](#collecting-feedback-via-ics)
+  - [Monomorphism vs. Polymorphism](#monomorphism-vs-polymorphism)
+  - [Feedback Lattice](#feedback-lattice)
+  - [Information Stored in Function Closures](#information-stored-in-function-closures)
+- [TurboFan](#turbofan)
+- [Speculative Optimization](#speculative-optimization)
+  - [`add` Example of Ignition and Feedback Vector](#add-example-of-ignition-and-feedback-vector)
+    - [Bytecode annotated](#bytecode-annotated)
+    - [Feedback Used To Optimize Code](#feedback-used-to-optimize-code)
+- [Deoptimization](#deoptimization)
+  - [Bailout](#bailout)
+    - [Example of x86 Assembly Code including Checks and Bailouts](#example-of-x86-assembly-code-including-checks-and-bailouts)
+  - [Lazy Cleanup of Optimized Code](#lazy-cleanup-of-optimized-code)
+  - [Deoptimization Loop](#deoptimization-loop)
+  - [Causes for Deoptimization](#causes-for-deoptimization)
+    - [Modifying Object Shape](#modifying-object-shape)
+      - [Considerations](#considerations)
+    - [Class Definitions inside Functions](#class-definitions-inside-functions)
+      - [Considerations](#considerations-1)
+      - [Resources](#resources-1)
+  - [Inlining Functions](#inlining-functions)
+- [Background Compilation](#background-compilation)
+- [Sea Of Nodes](#sea-of-nodes)
+  - [Advantages](#advantages)
+- [CodeStubAssembler](#codestubassembler)
+  - [What is the CodeStubAssember aka CSA?](#what-is-the-codestubassember-aka-csa)
+  - [Why is it a Game Changer?](#why-is-it-a-game-changer)
+    - [Improvements via CodeStubAssembler](#improvements-via-codestubassembler)
+- [Recommendations](#recommendations)
+- [Resources](#resources-2)
+  - [Slides](#slides)
+  - [Videos](#videos)
+  - [More Resources](#more-resources)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-# v8 Compiler
+## Goals
 
-## Components
+[watch](https://youtu.be/HDuSEbLWyOY?t=7m22s)
 
-### Base Compiler 
+> Speed up real world performance for modern JavaScript, and enable developers to build a
+> faster future web.
 
-- is used for all code initially
-- generates code quickly without heavy optimizations
-- compilation with the base compiler is very fast generates little code
+- fast startup vs. peak performance
+- low memory vs. max optimization
+- Ignition Interpreter allows to run code with some amount of optimization very quickly and has
+  very low memory footprint
+- TurboFan makes functions that run a lot fast, sacrificing some memory in the process
+- designed to support entire JavaScript language and make it possible to quickly add new
+  features and to optimize them fast and incrementally
```

Comment:
The new design with Ignition and TurboFan allows to integrate new language features very quickly and develop powerful optimizations incrementally.

Reply:
This is described in detail [here](https://github.com/thlorenz/v8-perf/blob/turbo-updates/compiler.md#new-language-features), not sure what we should add, but open to suggestions.